libevm/VM.cpp

```diff
@@ -42,6 +42,25 @@ template <class S> S modWorkaround(S const& _a, S const& _b)
 	return (S)(s512(_a) % s512(_b));
 }
 
+/// Implementation of EXP.
+
+/// This implements exponentiation by squaring algorithm.
+/// Is is faster than boost::multiprecision::powm() because avoids explicit
+/// mod operation.
+/// Do not inline it.
+u256 exp(u256 _base, u256 _exponent)
+{
+	using boost::multiprecision::limb_type;
+	u256 result = 1;
+	while (_exponent) {
+		if (static_cast<limb_type>(_exponent) & 1)
```

Comment:
It is faster, probably because & and == true is applied to native int instead of u256. But bit_test will be perfect here, thanks.